tests/ragger/test_sign.py

```diff
@@ -1,67 +1,372 @@
+                confirm_action = NavInsID.BOTH_CLICK
+                navigator.navigate_until_text_and_compare(next_action,
+                                                          [confirm_action],
+                                                          "Reject",
+                                                          ROOT_SNAPSHOT_PATH,
+                                                          test_name)
+            else:
+                instructions = [NavInsID.USE_CASE_REVIEW_TAP] * 2
+                instructions += [NavInsID.USE_CASE_CHOICE_REJECT,
+                                NavInsID.USE_CASE_CHOICE_CONFIRM,
+                                NavInsID.USE_CASE_STATUS_DISMISS]
+                navigator.navigate_and_compare(ROOT_SNAPSHOT_PATH,
+                                               test_name,
+                                               instructions)
+
+    except ExceptionRAPDU as e:
+        assert e.status == StatusWord.CONDITION_NOT_SATISFIED
+    else:
+        assert False  # An exception should have been raised
+
+
+def common_fail(backend: BackendInterface,
+                tx_params: dict,
+                expected: StatusWord,
+                path: str = BIP32_PATH):
+    app_client = EthAppClient(backend)
+
+    try:
+        with app_client.sign(path, tx_params):
+            pass
+
+    except ExceptionRAPDU as e:
+        assert e.status == expected
+    else:
+        assert False  # An exception should have been raised
+
+
 def test_legacy(firmware: Firmware, backend: BackendInterface, navigator: Navigator):
-    common(firmware, backend, navigator, {
+    tx_params: dict = {
         "nonce": NONCE,
         "gasPrice": Web3.to_wei(GAS_PRICE, "gwei"),
         "gas": GAS_LIMIT,
         "to": ADDR,
         "value": Web3.to_wei(AMOUNT, "ether"),
         "chainId": CHAIN_ID
-    })
+    }
+    common(firmware, backend, navigator, tx_params)
+
+
+# Legacy Zemu Send: Transfer amount >= 2^87 Eth on Ethereum app should fail
+def test_legacy_send_error(backend: BackendInterface):
+    tx_params: dict = {
+        "nonce": 38,
+        "gasPrice": 56775612312210000000001234554332,
+        "gas": GAS_LIMIT,
+        "to": ADDR3,
+        "value": 12345678912345678912345678000000000000000000,
+        "chainId": CHAIN_ID
+    }
+    common_fail(backend, tx_params, StatusWord.EXCEPTION_OVERFLOW, path=BIP32_PATH2)
+
+
+# Legacy Zemu Send BSC: Transfer bsc
+def test_legacy_send_bsc(firmware: Firmware,
+                         backend: BackendInterface,
+                         navigator: Navigator,
+                         test_name: str):
+    tx_params: dict = {
+        "nonce": 1,
+        "gasPrice": Web3.to_wei(GAS_PRICE2, 'gwei'),
+        "gas": GAS_LIMIT,
+        "to": ADDR2,
+        "value": 31415926913374232,
+        "chainId": 56
+    }
+    common(firmware, backend, navigator, tx_params, test_name, BIP32_PATH2)
+
+
+# Legacy Zemu ChainId: Transfer on network 112233445566 on Ethereum
```

Comment:
We probably can get rid of everything that precedes the `:` in those function comments.

Reply:
Fixed